nornir/core/inventory.py

```diff
@@ -460,3 +460,27 @@ def add_group(self, name: str, **kwargs) -> None:
             )
         }
         self.groups.update(group)
+
+    def get_inventory_dict(self) -> Dict:
+        """
+        Return serialized dictionary of inventory
+        """
+        return deserializer.inventory.Inventory.serialize(self).dict()
+
+    def get_defaults_dict(self) -> Dict:
+        """
+        Returns serialized dictionary of defaults from inventory
+        """
+        return self.get_inventory_dict().get("defaults")
+
+    def get_groups_dict(self) -> Dict:
+        """
+        Returns serialized dictionary of groups from inventory
+        """
+        return self.get_inventory_dict().get("groups")
+
+    def get_hosts_dict(self) -> Dict:
+        """
+        Returns serialized dictionary of hosts from inventory
+        """
+        return self.get_inventory_dict().get("hosts")
```

Comment:
I think we should call instead:

```
         return {k: deserializer.inventory.InventoryElement.serialize(v).dict() for k, v in self.hosts}
```

Reply:
I see what you're saying on this and the groups. It avoids calling an additional function to perform the action. One minor correction however, just adding `.items()` on both this and groups
```
        return {k: deserializer.inventory.InventoryElement.serialize(v).dict() for k, v in self.hosts.items()}
```